### DISCUSSION TITLE
Why is passing (char)<value> to Assert.That() make unit test not runnable?

### DISCUSSION THREAD

**Discussion by supawichza40** [nunit/nunit#4023] [Q&A] [answered]
Hi everyone, I am really new to the community and if I cause any problem apologise in advance, and would be happy to hear your feedback!
Basically I was doing a unit testing for my own personal project, but then when I passed in (char)5 and compared to value 5, the unit test was not running anymore, then when I swap back, to int 5 and int 5, it then work again.
I tried with 'c' as an input and it does work, but when I tried (char)5, it does not run anymore even though it the same type System.Char.
![unittestNotRunning](https://user-images.githubusercontent.com/88581422/146683530-1b06af97-b014-4aa7-b651-e72d7df95f00.PNG)

This is the code I used for Unit test.
![codewitherror](https://user-images.githubusercontent.com/88581422/146683694-0580b2b7-6308-4ec5-9ef4-bf65f26f93dc.PNG)
Every help and hint would be greatly appreciate.

**Comment by mikkelbu** [comment, marked as answer]
I can run the test using the console-runner, so I guess it is a problem with the interaction between the framework and the adapter.

I'm a bit unsure if the `PassTwo(5,'\u0005')` confuses the adapter (it didn't work for me with `Include="NUnit3TestAdapter" Version="4.0.0"`. CC @OsirisTerje

**Reply by mikkelbu** [reply]
If I use a runsettings - like below - then it works
```
<?xml version="1.0" encoding="utf-8"?>
<RunSettings>
  <NUnit>
     <UseParentFQNForParametrizedTests>True</UseParentFQNForParametrizedTests>
     <UseNUnitIdforTestCaseId>True</UseNUnitIdforTestCaseId>
  </NUnit>

</RunSettings>
```

**Reply by supawichza40** [reply]
So this error is specifically for the VS that NUnit is running under since the VS is running both the Adapter and framework, is there a way for fixing this if you are running the NUnit in Visual Studio that installed the Test Adapter.
Thank you @mikkelbu for clarifying, since I thought it might be my computer that cause this problem.

**Comment by manfred-brands** [comment]
It might just be a display issue.
The code tries to display `(char)5` which is not a printable character. You said it worked with 'c' and @mikkelbu made it work when using a different naming option.
